src/main/java/de/tum/in/www1/artemis/web/rest/AccountResource.java

```diff
@@ -187,6 +187,22 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
         userService.changePassword(passwordChangeDto.getCurrentPassword(), passwordChangeDto.getNewPassword());
     }
 
+    /**
+     * {@code POST  /account/change-language} : changes the current user's language key.
+     *
+     * @param languageKey languageKey to change to.
+     * @throws IllegalArgumentException {@code 400 (Bad Request)} if the language key is not 'en' or 'de'.
+     */
+    @PostMapping(path = "/account/change-language")
+    public void changeLanguageKey(@RequestBody String languageKey) {
+        User user = userRepository.getUser();
+        String langKey = languageKey.replaceAll("\"", "").toLowerCase().trim();
+        if (!"en".equals(langKey) && !"de".equals(langKey)) {
+            throw new BadRequestAlertException("Invalid language key", "Account", "invalidLanguageKey");
```

Comment:
@throws IllegalArgumentException seams to be wrong, since you throw another exception.
Additionally, can you also include the invalid key in the error message of the exception?

Reply:
Jup, I also think this should be a `IllegalArgumentException`. The user will not see the alert anyway, if the client behaves correctly.